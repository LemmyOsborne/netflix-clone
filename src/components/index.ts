export { Jumbotron } from "./Jumbotron/Jumbotron";
export { Footer } from "./Footer/Footer";