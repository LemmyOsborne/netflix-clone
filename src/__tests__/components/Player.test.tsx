import React from "react";
import { Player } from "../../components";
import { render, fireEvent } from "@testing-library/react";

describe("<Player/>", () => {
    it("renders the <Player/> with Witcher trailer video", () => {
        const { container, getByText, queryByTestId } = render(
            <Player>
            <Player.Video src="/videos/trailers/witcher.mp4" />
            <Player.Button>Play</Player.Button>
        </Player>
        )

        expect(queryByTestId("player")).toBeFalsy()
        fireEvent.click(getByText("Play"))
        expect(queryByTestId("player")).toBeTruthy()
        //@ts-ignore
        fireEvent.click(queryByTestId("player"))
        expect(queryByTestId("player")).toBeFalsy()
        expect(container.firstChild).toMatchSnapshot()
    })
})