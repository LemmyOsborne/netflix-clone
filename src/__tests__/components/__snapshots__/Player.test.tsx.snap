// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player/> renders the <Player/> with Witcher trailer video 1`] = `
<div
  class="sc-jQrDum"
>
  <button
    class="sc-GEbAx eTYOrQ"
  >
    Play
  </button>
</div>
`;
